lib/api/CInferenceModelDefinition.cc

```diff
@@ -113,6 +128,29 @@ void addJsonArray(const std::string& tag,
     }
     writer.addMember(tag, array, parentObject);
 }
+
+template<typename T>
+void addJsonArray(const std::string& tag,
+                  const std::vector<T>& vector,
+                  CSerializableToJson::TGenericLineWriter& writer) {
+    writer.Key(tag);
+    rapidjson::Value array{writer.makeArray(vector.size())};
+    for (const auto& value : vector) {
+        array.PushBack(toJson(value), writer.getRawAllocator());
+    }
+    writer.write(array);
+}
+
+void addJsonArray(const std::string& tag,
+                  const std::vector<std::string>& vector,
+                  CSerializableToJson::TGenericLineWriter& writer) {
+    writer.Key(tag);
+    rapidjson::Value array{writer.makeArray(vector.size())};
+    for (const auto& value : vector) {
+        array.PushBack(toJson(value, writer), writer.getRawAllocator());
+    }
+    writer.write(array);
```

Comment:
This is still copying the text in all the strings twice.  You can just copy it once like this:

```suggestion
    writer.StartArray();
    for (const auto& value : vector) {
        writer.String(value);
    }
    writer.EndArray();
```

Reply:
Fixed